### PR TITLE
semgrep 0.17.0

### PR DESCRIPTION
**Pull request by brendongo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/semgrep.rb [hunk 1]
```diff
@@ -89,6 +89,9 @@ class Semgrep < Formula
     sha256 "91056c15fa70756691db97756772bb1eb9678fa585d9184f24534b100dc60f4a"
   end
 
+  # Remove Sudo Command in install script. Safe to remove patch on 0.18.0
+  patch :DATA
+
```
Comment (SeekingMeaning) [inline Formula/semgrep.rb:94]: I might instead recommend adding this to `install`:

```ruby
# Remove Sudo Command in install script. Safe to remove patch on 0.18.0
inreplace "ocaml-tree-sitter/scripts/install-tree-sitter-lib", "sudo make install", "make install"
```

File: Formula/semgrep.rb [hunk 2]
```diff
@@ -148,3 +151,18 @@ def silly_eq(a, b):
...
++++ b/ocaml-tree-sitter/scripts/install-tree-sitter-lib
+@@ -18,7 +18,7 @@ fi
+   cd tree-sitter
+   git clean -dfX
+   make
+-  sudo make install
++  make install
+
+   # Ensure libtree-sitter is found at linking time.
+   # MacOS doesn't have ldconfig. Maybe it works without it?
```
Comment (SeekingMeaning) [inline Formula/semgrep.rb:168]: See above comment

**Comment by SeekingMeaning** [comment]
Reference: returntocorp/ocaml-tree-sitter#83

**Comment by brendongo** [comment]
Made suggested changes

**Comment by brendongo** [comment]
@SeekingMeaning I don't think inreplace works?

**Comment by brendongo** [comment]
Looks like it has to be in def install

**Review by SeekingMeaning — APPROVED** [review]

**Comment by brendongo** [comment]
Thanks @SeekingMeaning!

**Pull request closed without merging**